src/plugins/npcs/varrock/blue-moon-inn-plugin.ts

```diff
@@ -0,0 +1,168 @@
+import { ActionType, RunePlugin } from "@server/plugins/plugin";
+import { npcIds } from "@server/world/config/npc-ids";
+import { npcAction } from "@server/world/actor/player/action/npc-action";
+import { dialogue, Emote, execute } from "@server/world/actor/dialogue";
+import { itemIds } from "@server/world/config/item-ids";
+import { widgets } from "@server/world/config/widget";
+
+
+const talkToBartender : npcAction = (details) => {
+    const { player, npc } = details;
+
+    dialogue([player, { npc, key: 'bartender'}], [
+        bartender => [Emote.HAPPY, 'What can I do yer for?'],
+        options => [
+            `A glass of your finest ale please.`, [
+                player => [Emote.HAPPY, `A glass of your finest ale please.`],
+                bartender => [Emote.HAPPY, `No problemo. That'll be 2 coins.`],
+                execute(() => {
+                    const index = player.inventory.findIndex(itemIds.coins);
+                    const hasCoins = player.inventory.has(itemIds.coins);
+
+                    if (!hasCoins) {
+                        dialogue([player, {npc, key: 'bartender'}], [
+                            player => [Emote.VERY_SAD, `Oh dear. I don't seem to have enough money.`],
+                        ]);
+                    }
+
+                    if (hasCoins && (player.inventory.amountInStack(index) >= 2)) {
+
+                        const amount = player.inventory.amountInStack(index);
+                        // Check inventory.
+                        if (!player.inventory.hasSpace()) {
+                            player.sendMessage(`You don't have enough space in your inventory.`);
```

Comment:
Could we move this to the player object as a public function?

Reply:
That sounds like work - I just want beer :(